src/data/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,73 @@
+import { meRequest } from '@/api'
+import { User } from '@/types/entity/User'
+import { createContext, useReducer } from 'react'
+
+type stateType = {
+  isAuthenticated: boolean
+  user?: User
+}
+
+const initialState: stateType = {
+  isAuthenticated: false,
+  user: undefined
+}
+
+export const AuthContext = createContext({
+  ...initialState,
+  login: (user: User) => {},
+  logout: () => {},
+  me: () => {}
+})
+
+const reducer = (state: stateType, action: { type: string; payload: any }) => {
+  switch (action.type) {
+    case 'LOGIN':
+      return {
+        ...state,
+        user: action.payload,
+        isAuthenticated: true
+      }
+    case 'LOGOUT':
+      return {
+        ...state,
+        user: undefined,
+        isAuthenticated: false
+      }
+    default:
+      return state
+  }
+}
+
+const AuthContextProvider = ({ children }: { children: React.ReactNode }) => {
+  // #region Reducer ====================================================================
+  const [state, dispatch] = useReducer(reducer, initialState)
+  // #endregion
+
+  // #region Setter Functions ===========================================================
+  const login = (user: User) => {
+    dispatch({ type: 'LOGIN', payload: user })
+  }
+
+  const logout = () => {
+    dispatch({ type: 'LOGOUT', payload: null })
+  }
+
+  // #endregion
+
+  // #region Helper Functions =============================================================
+  const me = async () => {
+    try {
+      console.log('FILTERED 2')
```

Comment:
unnecessaryy debug log here, maybe we can disable the console logs in production build like this in vite.config.ts

`
    esbuild: {
      pure: mode === "production" ? ["console.log"] : [],
    },
`